src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -345,6 +345,16 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
         }
     }
 
+    // If editing an existing train dialog, return its Id, otherwise null
+    sourceTrainDialogId(): string | null {
```

Comment:
Would this also be used in LogDialogs here?
https://github.com/Microsoft/ConversationLearner-UI/blob/master/src/routes/Apps/App/LogDialogs.tsx#L636

Seems like it is not specific to train dialogs

Reply:
Log Dialogs should never be in the "NEW" state as they always start with an existing log.  In fact they should never be in "BRANCH" either so that part of the check can go away

Reply:
Was able to simplify this quite a bit